mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -61,6 +65,85 @@ static bool isConstantZero(Value v) {
   return false;
 }
 
+static bool isNegInfAttribute(Attribute value) {
+  if (auto fpValue = dyn_cast<FloatAttr>(value)) {
+    auto value = fpValue.getValue();
+
+    std::pair<APFloat, llvm::detail::opStatus> floatRes = rock::createAPFloat(
+        fpValue.getType(), -std::numeric_limits<float>::infinity());
+    auto expectedValue = floatRes.first;
+    auto status = floatRes.second;
+    assert(status == APFloat::opOK);
+
+    return value.compare(expectedValue) == llvm::APFloat::cmpEqual;
+  }
+  if (auto splatValue = dyn_cast<SplatElementsAttr>(value))
+    return isNegInfAttribute(splatValue.getSplatValue<Attribute>());
+  if (auto elementsValue = dyn_cast<ElementsAttr>(value))
+    return llvm::all_of(elementsValue.getValues<Attribute>(),
+                        isNegInfAttribute);
+  if (auto elementsValue = dyn_cast<DenseElementsAttr>(value))
+    return llvm::all_of(elementsValue.getValues<Attribute>(),
+                        isNegInfAttribute);
+  if (auto arrayValue = dyn_cast<ArrayAttr>(value))
+    return llvm::all_of(arrayValue.getValue(), isNegInfAttribute);
+
+  return false;
+}
+
+static bool isConstIsNegInf(Value v) {
+  if (auto cst = v.getDefiningOp<arith::ConstantOp>())
+    return isNegInfAttribute(cst.getValue());
+  if (auto cst = v.getDefiningOp<tosa::ConstOp>())
+    return isNegInfAttribute(cst->getAttr("value"));
+  return false;
+}
+
+static bool isConstRangeAttribute(Attribute value, int32_t expectedVal) {
+  if (auto intValue = dyn_cast<IntegerAttr>(value)) {
+    auto value = intValue.getValue();
+
+    std::pair<APInt, bool> intRes =
+        rock::createAPInt(intValue.getType(), expectedVal);
+    auto expectedValue = intRes.first;
+    bool overflow = intRes.second;
+    if (overflow)
+      return false;
+
+    return expectedValue == value;
+  }
+  if (auto splatValue = dyn_cast<SplatElementsAttr>(value))
+    return false;
+  if (auto elementsValue = dyn_cast<ElementsAttr>(value))
+    return llvm::all_of(llvm::enumerate(elementsValue.getValues<Attribute>()),
+                        [](const auto &indexedAttr) {
+                          return isConstRangeAttribute(indexedAttr.value(),
+                                                       indexedAttr.index());
```

Comment:
Isn't it passing `indexedAttr.index()` as the expectedVal ? We want to compare against `expectedVal` passed to first recursive call